source/Core/Config.php

```diff
@@ -1068,13 +1068,17 @@ public function getShopCurrency()
      */
     public function getActShopCurrencyObject()
     {
-        $cur = $this->getShopCurrency();
-        $currencies = $this->getCurrencyArray();
-        if (!isset($currencies[$cur])) {
-            return $this->_oActCurrencyObject = reset($currencies); // reset() returns the first element
+        if ($this->_oActCurrencyObject === null) {
+            $cur = $this->getShopCurrency();
+            $currencies = $this->getCurrencyArray();
+            if (!isset($currencies[$cur])) {
+                $this->_oActCurrencyObject = reset($currencies); // reset() returns the first element
```

Comment:
Please check the coding standards. "_" and "hungarian" notations are not allowed in new code anymore.

Reply:
@MaxBoeh, you are right, sorry, haven't noticed. I thought it is some new one for caching. Thanks for clarifying.